### DISCUSSION TITLE
Run time setting of MTU size

### DISCUSSION THREAD

**Discussion by bbbakke** [zephyrproject-rtos/zephyr#57149] [Q&A]
Looking at the current gatt api, the only public function I found to set MTU size for a connection is bt_gatt_exchange_mtu. This api takes a param struct that only contains a callback and uses the max MUT size from the build time parameters. However, what I would like is provide the MTU size for the given connection (guessing it should be added to the param struct) that I can set based on the given service of the connection. 

Why: I connect to multiple services. One of them needs to transfer large packets of data (I want to set the MTU size to 247) while the other service doesn't need packets nearly that large (I what to set the MTU size to 95 bytes for that service).

Question: is there an API available that I can call to set the MTU to a specific size for a give connection handle? 

For reference, I am developing on a custom board with a nRF5340 and have been using the nRF5340DK for some of the development.

**Comment by hermabe** [comment]
There is no API for this, the GATT MTU is determined by the buffer size kconfigs.

Is the idea here that you want to allocate more smaller buffers for some connections? Currently all ACL data buffers are the same size and allocated from the same pool.
